client/extensions/wp-job-manager/state/data-layer/settings/index.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Internal dependencies
+ */
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { displaySettings, enableSettings } from '../../settings/actions';
+import { WP_JOB_MANAGER_FETCH_SETTINGS } from 'wp-job-manager/state/action-types';
+
+export const fetch = ( { dispatch }, action ) => {
+	const { siteId } = action;
+
+	dispatch( http( {
+		method: 'GET',
+		path: `/jetpack-blogs/${ siteId }/rest-api/`,
+		query: {
+			path: '/wpjm/v1/settings',
+		},
+	}, action ) );
+};
+
+export const displayAndEnable = ( { dispatch }, { siteId }, next, { data } ) => {
+	dispatch( displaySettings( siteId, data ) );
+	dispatch( enableSettings( siteId ) );
+};
+
+export const enable = ( { dispatch }, { siteId } ) => dispatch( enableSettings( siteId ) );
+
+const dispatchSettingsRequest = dispatchRequest( fetch, displayAndEnable, enable );
```

Comment:
I'm having a bit of a problem grokking the nomenclature here. `displaySettings` (and `WP_JOB_MANAGER_DISPLAY_SETTINGS`) sounds very UI related, but we're really just updating Redux state thru the reducer, aren't we? Would maybe `updateSettings` be a better fit?

I also find `enableSettings` kind of confusing. If I read the code right, you chose that name because `WP_JOB_MANAGER_ENABLE_SETTINGS` is used to signal that settings have been fetched, and any intermediary state (during which controls are disabled) is over. Is my reading correct? If yes, I'd say that's again too UI centric for something on the data layer.

To achieve the same thing maybe somewhat more semantically, can we drop `enableSettings` (and `WP_JOB_MANAGER_ENABLE_SETTINGS`) for now, and have the `fetching` reducer listen to `WP_JOB_MANAGER_DISPLAY_SETTINGS` (or `_UPDATE_SETTINGS`) instead?

For the error case, let's stick with a `_FETCH_ERROR` for now. Goes a bit against `data-layer` recommendations, but so does a `fetching` state, which is what we'd need that action name constant mostly for 😄